src/main/java/net/openhft/chronicle/queue/impl/single/StoreTailer.java

```diff
@@ -1017,6 +1015,17 @@ public ExcerptTailer originalToEnd() {
 
     }
 
+    private boolean originalToEndLoopCondition(long approximateLastIndex, long index) {
+        if (direction == FORWARD) {
+            return true;
+        } else if (direction == BACKWARD) {
+            // Do not let index run past the approximate last index
+            return index < approximateLastIndex;
+        } else {
+            return true;
+        }
```

Comment:
The whole method can be simplified as `if(condition) return smth`. i.e. dropping all `else`.

Reply:
I agree. I would usually change this, however the code is already integrated into `develop` in this form and we are now backporting to the 2.23 release. I'll be revisiting the code in develop so will address there.